cmd/ctr/commands/snapshots/snapshots.go

```diff
@@ -271,6 +281,115 @@ var removeCommand = cli.Command{
 	},
 }
 
+var createCommand = cli.Command{
+	Name:      "create",
+	Usage:     "create a new snapshot from an image",
+	ArgsUsage: "[flags] <image name>",
+	Description: `Creates a new snapshot from a specified image.
+
+When you're done, use the remove command.
+`,
+	Flags: append(commands.SnapshotterFlags,
+		cli.BoolFlag{
+			Name:  "print-mounts",
+			Usage: "Print the snapshotter layer mounts",
+		},
+		cli.BoolFlag{
+			Name:  "print-chainid",
+			Usage: "Print the snapshot's chain ID",
+		},
+		cli.StringFlag{
+			Name:  "platform",
+			Usage: "Create the snapshot for the specified platform",
+			Value: platforms.DefaultString(),
+		},
+		cli.StringFlag{
+			Name:  "key",
+			Usage: "Optional key to use for snapshot. Default is image name.",
+		},
+	),
+	Action: func(context *cli.Context) (retErr error) {
+		imageName := context.Args().First()
+		if imageName == "" {
+			return fmt.Errorf("please provide an image to create a snapshot for")
+		}
+
+		key := context.String("key")
+		if key == "" {
+			key = imageName
+		}
+
+		client, ctx, cancel, err := commands.NewClient(context)
+		if err != nil {
+			return err
+		}
+		defer cancel()
+
+		snapshotter := context.String("snapshotter")
+		if snapshotter == "" {
+			snapshotter = containerd.DefaultSnapshotter
+		}
+
+		ps := context.String("platform")
+		p, err := platforms.Parse(ps)
+		if err != nil {
+			return errors.Wrapf(err, "unable to parse platform %s", ps)
+		}
+
+		ctx, done, err := client.WithLease(ctx,
+			leases.WithID(imageName),
+			leases.WithExpiration(24*time.Hour),
+			leases.WithLabels(map[string]string{
+				"containerd.io/gc.ref.snapshot." + snapshotter: key,
+			}),
+		)
+		if err != nil && !errdefs.IsAlreadyExists(err) {
+			return err
+		}
+
+		defer func() {
+			if retErr != nil && done != nil {
+				done(ctx)
+			}
+		}()
+
+		img, err := client.ImageService().Get(ctx, imageName)
+		if err != nil {
+			return err
+		}
+
+		i := containerd.NewImageWithPlatform(client, img, platforms.Only(p))
+		if err := i.Unpack(ctx, snapshotter); err != nil {
+			return errors.Wrap(err, "error unpacking image")
+		}
+
+		diffIDs, err := i.RootFS(ctx)
+		if err != nil {
+			return err
+		}
+		chainID := identity.ChainID(diffIDs).String()
+		if context.Bool("print-chainid") {
+			fmt.Println(chainID)
+		}
+
+		s := client.SnapshotService(snapshotter)
+		mounts, err := s.Prepare(ctx, key, chainID)
```

Comment:
I think we should provide the option of creating a read-only (View) snapshot too. We can add a `--read-only` flag for the command.

Reply:
Done :)